systems/subscriber/subscriber-registry/pkg/server/subscriber.go

```diff
@@ -0,0 +1,288 @@
+			Address:               req.GetAddress(),
+			ProofOfIdentification: req.GetProofOfIdentification(),
+			DateOfBirth:           birthday.GoString(),
+			IdSerial:              req.GetIdSerial()},
+	}, nil
+
+}
+
+func (s *SubcriberServer) Delete(ctx context.Context, req *pb.DeleteSubscriberRequest) (*pb.DeleteSubscriberResponse, error) {
+	subscriberID := req.GetSubscriberID()
+	if subscriberID == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "subscriberID must not be empty")
+	}
+
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+	subscriberIDUUID, err := uuid.FromString(subscriberID)
+
+	logrus.Infof("Delete Subscriber : %v ", subscriberID)
+	err = s.subscriberRepo.Delete(subscriberIDUUID)
+	if err != nil {
+		logrus.WithError(err).Error("error while deleting subscriber")
+		return nil, grpc.SqlErrorToGrpc(err, "subscriber")
+	}
+	route := s.subscriberRoutingKey.SetAction("delete").SetObject("subscriber").MustBuild()
+	err = s.msgbus.PublishRequest(route, req)
+	if err != nil {
+		logrus.Errorf("Failed to publish message %+v with key %+v. Errors %s", req, route, err.Error())
+	}
+	if err != nil {
+		logrus.Errorf("Failed to publish message %+v with key %+v. Errors %s", req, route, err.Error())
+	}
+	return &pb.DeleteSubscriberResponse{
+		SubscriberID: subscriberID,
+	}, nil
+}
+
+func (s *SubcriberServer) Get(ctx context.Context, req *pb.GetSubscriberRequest) (*pb.GetSubscriberResponse, error) {
+
+	subscriberID := req.GetSubscriberID()
+	if subscriberID == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "subscriberID must not be empty")
+	}
+
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+
+	subscriberIDUUID, err := uuid.FromString(subscriberID)
+	logrus.Infof("GetSubscriber : %v ", subscriberID)
+	subscriber, err := s.subscriberRepo.Get(subscriberIDUUID)
+
+	if err != nil {
+		logrus.Error("error getting a subscriber" + err.Error())
+
+		return nil, grpc.SqlErrorToGrpc(err, "subscriber")
+	}
+
+	resp := &pb.GetSubscriberResponse{Subscriber: dbSubscriberToPbSubscribers(subscriber)}
+
+	return resp, nil
+
+}
+
+func (s *SubcriberServer) ListSubscribers(ctx context.Context, req *pb.ListSubscribersRequest) (*pb.ListSubscribersResponse, error) {
+
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+
+	logrus.Infof("List all subscribers")
+
+	subscribers, err := s.subscriberRepo.ListSubscribers()
+	if err != nil {
+		logrus.WithError(err).Error("error while getting all subscribers")
+		return nil, grpc.SqlErrorToGrpc(err, "subscribers")
+	}
+
+	subscriberList := &pb.ListSubscribersResponse{
+		Subscribers: dbsubscriberToPbSubscribers(subscribers),
+	}
+
+	return subscriberList, nil
+}
+func (s *SubcriberServer) GetByNetwork(ctx context.Context, req *pb.GetByNetworkRequest) (*pb.GetByNetworkResponse, error) {
+	networkID := req.GetNetworkID()
+	if networkID == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "networkID must not be empty")
+	}
+
+	if err := ctx.Err(); err != nil {
```

Comment:
same as above

Reply:
I have added comment earlier in this thread.